### PR TITLE
feat: add debug logging to investigate missing CSOD deletes

### PR DESCRIPTION
**Pull request by johnnagro**
## Description

I've narrowed down where we seem to be losing the content delete records - this adds additional debug logging.

## References

- https://2u-internal.atlassian.net/browse/ENT-6081
- #1627

### CODE REVIEW

File: tests/test_integrated_channels/test_utils.py [hunk 1]
```diff
@@ -309,3 +309,25 @@ def test_truncate_item_collections(self):
         assert len(out_a) == 3
         assert len(out_b) == 5
         assert len(out_c) == 2
+
+        in_a = []
```
Comment (johnnagro) [inline tests/test_integrated_channels/test_utils.py:313]: i wanted to add some test cases where the first couple buckets are small or empty to make sure we'd still get the delete content (bucket c)

**Review by kiram15 — APPROVED** [review]
Good luck!
